### ISSUE TITLE
Missing DSA_meth_set1_name

### ISSUE THREAD

**Issue by ffontaine** [libressl-portable/openbsd#130]
There is no `DSA_meth_set1_name` as a result the following build failure is raised with pkcs11-helper since version 3.5.0 and https://github.com/libressl-portable/openbsd/commit/62c7bff5397fa44f595b161cd593d9456eca236e:

```
pkcs11h-openssl.c: In function 'DSA_meth_set1_name':
pkcs11h-openssl.c:239:41: error: invalid use of incomplete typedef 'DSA_METHOD' {aka 'struct dsa_method'}
  239 |  rv = _pkcs11h_mem_strdup ((void *)&meth->name, name);
      |                                         ^~

```

Would it be possible to add this function? `RSA_meth_set1_name` is available since version 2.8.1 and https://github.com/libressl-portable/openbsd/commit/f23a529705e8dc36ad12e26aaa37944a52f1d678.

Full build log: http://autobuild.buildroot.org/results/9b0d3bf7d97696c7be6de1724daaba196626b865

**Comment by botovq** [comment]
Thanks for the report and apologies for the late reply. I have added
DSA_meth_set1_name() and DSA_meth_get0_name(). They will be available
in LibreSSL 3.6.0 and later.

**Comment by 4a6f656c** [comment]
This API is available in LibreSSL 3.6.0.
